### PR TITLE
Always respond with HTTP 206 for range requests

### PR DESCRIPTION
**Pull request by gaul**
This matches the GCS behavior.

### CODE REVIEW

File: fakestorage/object.go [hunk 1]
```diff
@@ -516,7 +516,7 @@ func (s *Server) downloadObject(w http.ResponseWriter, r *http.Request) {
 
 	status := http.StatusOK
 	start, end, content := s.handleRange(obj, r)
-	if len(content) != len(obj.Content) {
+	if r.Header.Get("Range") != "" {
```
Comment (fsouza) [inline fakestorage/object.go:519]: Let's have `handleRange` return whether or not this is a "ranged" request, this way only that method is dealing with request headers for range.

Comment (gaul) [reply fakestorage/object.go:519]: Done.

**Review by fsouza — APPROVED** [review]

**Comment by fsouza** [comment]
Change looks good. Can you fix the test?

**Comment by gaul** [comment]
> Change looks good. Can you fix the test?

Done.

**Review by fsouza — APPROVED** [review]

**Comment by gaul** [comment]
Combined with #510 now gcsfuse to read and write files.  fake-gcs-server integration will allow us to set up offline integration tests and CI.  Thanks!

**Pull request merged**